### DISCUSSION TITLE
How does Powertools handle fifo queues?

### DISCUSSION THREAD

**Discussion by whoDoneItAgain** [aws-powertools/powertools-lambda-python#1913] [Q&A] [answered]
I've been reading over the documentation and I cant find an answer.

Does Powertools stop processing items after a failed and marks the failed plus all subsequent as failed in its response back to SQS?

Ex. r1,r2,r3,r4,r5 is sent. function fails on r3. ReportBatchItemFailures includes r3 as well as r4 and r5.

**Comment by heitorlessa** [comment]
Not currently, it's a feature gap.

Could you open up a feature request so we can prioritise it, please?

I can think of a new specialised class to handle FIFO and any special traits in the future. If you'd also like to give this a try, please feel free to send a PR and we can co-author it to get it scheduled for our next release.

---

Revisiting the official Lambda/SQS docs, it does calls this portion out - though there might be instances that ordering/duplicates may not be guaranteed:

> If you're using this feature with a FIFO queue, your function should stop processing messages after the first failure and return all failed and unprocessed messages in batchItemFailures. This helps preserve the ordering of messages in your queue.

https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html

Thank you!!

**Comment by rubenfonseca** [comment, marked as answer]
I've converted this discussion into an [issue](https://github.com/awslabs/aws-lambda-powertools-python/issues/1927) so we can track it and prioritize it.

**Reply by heitorlessa** [reply]
Quick update @whoDoneItAgain - we're nearly finishing the implementation now. This will be available in the next release for you: https://github.com/awslabs/aws-lambda-powertools-python/pull/1934

**Reply by whoDoneItAgain** [reply]
That’s great news! 

As a stop gap I implemented a global bool variable that, when my code errors, it set it to true. The fist thing the record handler does is check that variable and if true immediately fails. This obviously not a great fix and not something that would fix the model of this project, but it was a quick and easy fix for my needs. 

Looking forward to the official release.
